src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs

```diff
@@ -371,7 +371,7 @@ public static void FileSystemWatcher_Path()
         watcher.Path = "..";
         Assert.Equal("..", watcher.Path);
 
-        string currentDir = Path.GetFullPath(".").Trim('.', Path.DirectorySeparatorChar);
+        string currentDir = Path.GetFullPath(".").TrimEnd('.', Path.DirectorySeparatorChar);
         watcher.Path = currentDir;
         Assert.Equal(currentDir, watcher.Path);
 
```

Comment:
On line 386, do you want to change currentDirRelative to use Path.DirectorySeparatorChar?